mlperf_logging/repo_checker/repo_checker.py

```diff
@@ -4,60 +4,101 @@
 import subprocess
 
 
-def _check_bad_filenames(submission_dir):
+def list_files_recursively(*path):
+    path = os.path.join(*path)
+    return [
+        os.path.join(dirpath, file)
+        for dirpath, dirs, files in os.walk(path)
+        for file in files
+    ]
+
+
+def list_dirs_recursively(*path):
+    path = os.path.join(*path)
+    return [dirpath for dirpath, dirs, files in os.walk(path)]
+
+
+def _check_bad_filenames(files, dirs):
     """Checks for filename errors.
     Git does not like filenames with spaces or that start with ., or /. .
     """
-    logging.info('Running git-unfriendly file name checks.')
-    names = [
-        os.path.join(dirpath, filename)
-        for dirpath, _, filenames in os.walk(submission_dir)
-        for filename in filenames
-        if filename.startswith(".") or "/." in filename or " " in filename
+    logging.info("Running git-unfriendly file name checks.")
+    dir_names = [(dir_, dir_.split("/")[-1]) for dir_ in dirs]
```

Comment:
maybe can use os.path.basename() here so the script can work on different os?

Reply:
Updated